### PR TITLE
Map data terrain deformation #18

### PR DESCRIPTION
**Pull request by Robalian**

### CODE REVIEW

File: Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs [hunk 1]
```diff
@@ -61,56 +63,52 @@ private void OnMapData(JSONObject data)
...
                     SpawnRoomObjects<MineralView>(data[key].list, MineralView.Path);
-                }
+
                 else if (key.Equals("w"))
-                {
                     SpawnRoomObjects<WallView>(data[key].list, WallView.Path);
-                }
+
                 else if (key.Equals("pb"))
-                {
```
Comment (thmsndk) [inline Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs]: I prefer the block statements tbh :)

File: Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs [hunk 2]
```diff
@@ -61,56 +63,52 @@ private void OnMapData(JSONObject data)
...
                     SpawnRoomObjects<PowerBankView>(data[key].list, PowerBankView.Path);
-                }
-                else if (key.Equals("p"))
-                {
-                    SpawnRoomObjects<PortalView>(data[key].list, PortalView.Path);
```
Comment (thmsndk) [inline Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs]: Removed portal support?

File: Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs [hunk 3]
```diff
@@ -61,56 +63,52 @@ private void OnMapData(JSONObject data)
...
-                }
             }
         }
 
-        private void SpawnRoomObjects<T>(List<JSONObject> list, string prefabPath)
-            where T : IMapViewComponent
+        private void SpawnRoomObjects<T>(List<JSONObject> list, string PrefabPath)
```
Comment (thmsndk) [inline Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs]: Should not be capital P in a variable name :P

**Review by thmsndk — CHANGES_REQUESTED** [review]
Why was PlaneDeformer.cs  removed?

Also looks like you removed portal "p" support  in mapview :)

File: Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs (hunk 2 above)
Comment (Robalian) [reply Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs]: merge mistake D:

File: Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs (hunk 1 above)
Comment (Robalian) [reply Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs]: I find adding 2 lines with brackets for 1 line of code over the top ;] But let's keep it consistent

File: Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs (hunk 3 above)
Comment (Robalian) [reply Assets/Scripts/Screeps3D/Rooms/Views/MapView.cs]: noted

**Comment by Robalian** [comment]
I needed wall deformation for SKs, but no swamp deformations, so instead of adding more arguments for plane deformer I moved both deformers into TerrainView. So now all terrain generation is in one place. If at some point we change walls/swamps to use different grids then PlaneDeformer could become quite messy.

**Comment by thmsndk** [comment]
Ahhh okay, that makes sense in regards to plane deformation

I noticed a large chunk was removed from Prefabs/RoomView.prefab,, unity being unity and all, are we missing anything inside RoomView that should still be there?

**Comment by Robalian** [comment]
just removed references to PlaneDeformer script, after pulling changes, so should be fine

**Review by thmsndk — APPROVED** [review]

**Pull request merged**